linux/hid.c

```diff
@@ -566,6 +734,52 @@ struct hid_device_info  HID_API_EXPORT *hid_enumerate(unsigned short vendor_id,
 					 * check for USB and Bluetooth devices above */
 					break;
 			}
+
+			/* Usage Page and Usage */
+			int res;
+			struct hidraw_report_descriptor rpt_desc;
+			int device_handle = open(dev_path, O_RDWR);
```

Comment:
can you check is `O_RDONLY` enough for this?

Reply:
O_RDONLY seems to be enough from my testing.